cmd/registry-replacer/main.go

```diff
@@ -600,6 +621,104 @@ func pruneReplacements(config *api.ReleaseBuildConfiguration, filter asDirective
 	return utilerrors.NewAggregate(errs)
 }
 
+// pruneUnusedBaseImages uses the fully-resolved config to make sure an image is not used directly in  the config, or within any of the tests.
+// If it is not, then we prune it.
+func pruneUnusedBaseImages(config *api.ReleaseBuildConfiguration, resolvedConfig *api.ReleaseBuildConfiguration) error {
+	usedBaseImages := sets.NewString()
+	for _, step := range resolvedConfig.RawSteps {
+		switch {
+		case step.BundleSourceStepConfiguration != nil:
+			getBundleSourceImages(&usedBaseImages, step.BundleSourceStepConfiguration)
+		case step.OutputImageTagStepConfiguration != nil:
+			usedBaseImages.Insert(string(step.OutputImageTagStepConfiguration.From))
+		case step.PipelineImageCacheStepConfiguration != nil:
+			usedBaseImages.Insert(string(step.PipelineImageCacheStepConfiguration.From))
+		case step.RPMImageInjectionStepConfiguration != nil:
+			usedBaseImages.Insert(string(step.RPMImageInjectionStepConfiguration.From))
+		case step.SourceStepConfiguration != nil:
+			usedBaseImages.Insert(string(step.SourceStepConfiguration.From))
+		case step.TestStepConfiguration != nil:
+			getTestStepImages(&usedBaseImages, step.TestStepConfiguration)
+
```

Comment:
return an error in the default case since it's a programming error not to check every possible type - i wish Go had better enum handling :|